src/qililab/result/stream_results.py

```diff
@@ -0,0 +1,84 @@
+from typing import Any
+
+import h5py
+import numpy as np
+
+
+class StreamArray:
```

Comment:
Finally, I want to ask for a change that is solely to improve the user experience.

This is something that I've been wanting to do for a long time now, and this is the perfect example to start doing it.

If you check popular python libraries (like numpy), you'll see that all of its classes/functions are shown in lowercase to the user, for example:
```pycon
>>> import numpy as np
>>> ar = np.array([1, 2])  # array is a class, but it is lowercase!
```

This is solely to improve user experience. The user shouldn't care if something is a class or not, thus they shouldn't care about lower/upper case letters.

In PennyLane, they solved this in a way that I really liked: they kept the syntax of upper-case letters for classes, but they created "constructors" (a.k.a functions) in lowercase that returned an instance of the class. In the PennyLane main `__init__` file, only these lower-case functions were imported (similar thing for documentation). [See an example in this link](https://github.com/PennyLaneAI/pennylane/blob/f25eb134d1aaca35c9cd886cd38f4283b457728e/pennylane/ops/op_math/pow.py#L39).

What do you think of doing the same thing? In this case it would be something like this:
```suggestion
def stream_results(shape: tuple, path: str, loops: dict[str, np.ndarray]):  # this is the constructor
    return StreamArray(shape=shape, path=path, loops=loops)

class StreamArray:
```

If we do this, the big chunk of the documentation (long explanation + examples) should be moved to this user-facing method, because this is what users will see. 😄 

Let me know if you have any questions!

Reply:
I like the idea actually and it we enforce all along qililab little by little I think it will indeed improve user experience.